http://dev.mutt.org/trac/changeset/5899%3Ad6f88fbf8387

Don't parse URL port as signed short, but to int and cast to unsigned
short. Closes #3264.

Index: url.c
===================================================================
--- url.c (revision 5845:0698e8195545)
+++ url.c (revision 5899:d6f88fbf8387)
@@ -144,7 +144,9 @@
   if ((p = strchr (t, ':')))
   {
+    int t;
     *p++ = '\0';
-    if (mutt_atos (p, (short*) &ciss->port) < 0)
+    if (mutt_atoi (p, &t) < 0 || t < 0 || t > 0xffff)
       return NULL;
+    ciss->port = (unsigned short)t;
   }
   else
